### ISSUE TITLE
[Musicxml Export] - Better way to show and hide staff

### ISSUE THREAD

**Issue by hhpmusic** [musescore/MuseScore#17398]
### Issue type

Other type of issue

### Steps to reproduce

When an orchestral score is set to hide empty staves, or when a piano score has temporary displayed Ossia staves, the exported Musicxml shows a confusing description of such staves. It only exports staff-distance, but in standard musicxml expression, it must be staff-details number=n print-object="yes" or"no". This is placed right at the show or hide point of the staff, either at the beginning of a system or the point a cutaway staff is show or hidden. The current way confuses not only the code reader like me, but also confuses braille music translation software.

### MuseScore Version

4.0.24.0.2

### Regression

No.

### Operating system

All

**Comment by lvinken** [comment]
Simple example illustrating the issue including incorrect (current) and correct (to be) output:
[issue_17398_example.zip](https://github.com/musescore/MuseScore/files/11438313/issue_17398_example.zip)

**Comment by Jojo-Schmitz** [comment]
I'm affraid I don't see how the correct.musicxml is (fully) correct, it looses a note in the voice staff, but indeed keeps the 'hide empty staves setting, the incorrect.musicxml just looses the hide empty staves setting.

Loosing a note at the cost of keeping that "hide empty staves" setting can't be the desired outcome, can it?

Mu3 keeps all notes, but looses the 'hide empty staves' setting, so is having the same problem this issue describes, but Mu4 seems to have an additional and much worse problem, in loosing notes. I guess the latter is worth a separate issue (and regression vs Mu3), isn't it?

**Comment by hhpmusic** [comment]
The example may have something missing. It's not mine example, but the thing I'm hoping is correct, thus when the staff is hidden, either throughout the whole system(s) or as cut-away, the staff-details should be shown instead of showing only staff-distance. The latter is very confusing, and I suggest not to use it when the staves are hidden. Missing notes is not a result of hiding staves.

> I'm affraid I don't see how the correct.musicxml is (fully) correct, on imporrt it looses a note in the voice staff, but indeed keeps the 'hide empty staves setting', the incorrect.musicxml just looses the 'hide empty staves' setting.
> 
> Loosing a note at the cost of keeping that 'hide empty staves' setting can't be the desired outcome, can it?
> 
> Mu3 keeps all notes, but looses the 'hide empty staves' setting, so is having the same problem this issue describes, but Mu4 seems to have an additional and much worse problem, in loosing notes. I guess the latter is worth a separate issue (and regression vs Mu3), isn't it? Or is that just a flaw in your sample files?

**Comment by Jojo-Schmitz** [comment]
>  Missing notes is not a result of hiding staves.

It of course shouldn't be but is in that example though

**Comment by hhpmusic** [comment]
Haha, he might mistakenly removed the note when manually editing the xml file.

**Comment by Jojo-Schmitz** [comment]
Apparently not, I checked the diff between the 2 musicxml files

**Comment by Jojo-Schmitz** [comment]
Isn't this related to #16135? Seems that is the broken import I'm talking about?

**Comment by hhpmusic** [comment]
So Ms actually doesn't correctly handle shoing/hiding staves in Musicxml at all. Staff-distance applies to the whole staff in a system, so using it for hiding staves is a wrong thought. Using staff-details print-object will handle staff appearance on-the-fly correctly, like a switch.

**Comment by hhpmusic** [comment]
No, it's different. Applying print-object to measure is not feasible, the standard is staff-details print-object syntax. If the staff is not printed, then it's hidden fully or partially during a system. If measure print-object is no, I think the output must be a blank space without anything. Don't print measure and remove staff space is a complicated solution than staff-details print-object="no".

> Is this issue addressed by #17035?

**Comment by lvinken** [comment]
Responding to the comments on my example. Just for your convenience it was meant to show how to correctly encode a temporarily hidden staff (in this case the first staff in the first system). To the best of my knowledge it is correct and it does import exactly as expected in Finale NotePad. I did not try to import it into MuseScore 4, doing that might have exposed bugs. Currently MuseScore does not handle the staff-details print-object=no on import.

**Comment by Jojo-Schmitz** [comment]
That's why I suggested the need of another issue for the import of these

**Comment by rettinghaus** [comment]
@hhpmusic Would you also need the "print-spacing" attribute to indicate cut-out scores?

**Comment by hhpmusic** [comment]
Yes, this is needed.
